src/main/kotlin/org/malachite/estella/services/HrPartnerService.kt

```diff
@@ -26,17 +42,48 @@ class HrPartnerService(
         return hrPartner
     }
 
-    fun updateHrPartner(id: Int, hrPartner: HrPartner) {
-        val currPartner: HrPartner = hrPartnerRepository.findByUserId(id).get()
+    fun updateHrPartner(id: Int, hrPartner: HrPartner, jwt: String?) {
+        if(!getPermissions(id, jwt).contains(Permission.UPDATE)) throw UnauthenticatedException()
+        updateHrPartner(id, hrPartner)
+    }
+
+    private fun updateHrPartner(id: Int, hrPartner: HrPartner) {
+        val currPartner: HrPartner = this.getHrPartner(id)
         val updated: HrPartner = currPartner.copy(
             organization = hrPartner.organization,
             user = hrPartner.user
         )
         hrPartnerRepository.save(updated)
-
     }
 
+    fun getHrsByOrganizationId(id: UUID?) =
+        this.getHrPartners()
+            .filter { it.organization.id == id }
 
+    private fun getHrsIdsByOrganizationId(id: UUID?) =
+        this.getHrsByOrganizationId(id)
+            .map { it.id }
 
-    fun deleteHrPartner(id: Int) = hrPartnerRepository.deleteById(id)
+    private fun deleteHrPartner(id: Int) = hrPartnerRepository.deleteById(id)
+
+    fun deleteHrPartner(id: Int, jwt: String?) {
+        if(!getPermissions(id, jwt).contains(Permission.DELETE)) throw UnauthenticatedException()
+        deleteHrPartner(id)
+    }
+
+    fun getPermissions(id: Int?, jwt: String?): Set<Permission> {
+        if(securityService.isCorrectApiKey(jwt)) return Permission.allPermissions()
+        val permissions = mutableSetOf(Permission.READ)
+        val user = securityService.getHrPartnerFromJWT(jwt) ?: securityService.getOrganizationFromJWT(jwt) ?: throw UnauthenticatedException()
+        if(user is HrPartner && user.id == id) {
+            permissions.addAll(Permission.allPermissions())
+        }
+        if(user is Organization && getHrsIdsByOrganizationId(user.id).contains(id) && user.verified) {
+            permissions.addAll(Permission.allPermissions())
+        }
+        if(user is Organization && id == null) {
+            permissions.add(Permission.CREATE)
+        }
+        return permissions
```

Comment:
To ogólnie rzecz biorąc wygląda źle, ale na chwilę obecną nie mam pomysłu jak to naprawić XD

Reply:
Ale dlaczego? Przechodzę po wszystkich możliwych uprawnieniach i po kolei sprawdzam, jakie im uprawnienia przysługują